medusa/restore_cluster.py

```diff
@@ -213,7 +214,7 @@ def _tokens_from_ringitem(ringitem):
 
         def _token_counts_per_host(tokenmap):
             for host, ringitem in tokenmap.items():
-                yield len(ringitem['tokens'])
+                return len(ringitem['tokens'])
```

Comment:
We don't actually need the `for` loop here after your update. We can just get the first host and return the length of its token list. I know the return breaks the loop, but it's probably a bit better to get the information directly.